src/components/Media/Iframe/Iframe.tsx

```diff
@@ -11,21 +13,91 @@ const b = block('media-component-iframe');
 
 const Iframe = (props: MediaComponentIframeProps) => {
     const {iframe, margins = true} = props;
-    const {height = 400, src, width, name, title} = iframe;
+    const {height = 400, src, width = '100%', name, title, justifyContent = 'center'} = iframe;
+
+    const formContainerRef = useRef<HTMLDivElement>(null);
+    const iframeRef = useRef<HTMLIFrameElement>();
+    const [iframeId] = useState(uuidv4());
+
+    const updateFormIframe = useCallback(
+        (container: HTMLDivElement) => {
+            if (iframeRef.current) {
+                iframeRef.current.src = src;
+            } else {
+                const iframeWidth = typeof width === 'number' ? `${width}px` : width;
+                let iframeHeight: string | number | undefined =
+                    typeof height === 'number' ? `${height}px` : height;
+
+                if (height === 'auto') {
+                    iframeHeight = undefined;
+                }
+
+                iframeRef.current = document.createElement('iframe');
+                iframeRef.current.src = src;
+                iframeRef.current.id = iframeId;
+                iframeRef.current.name = name || iframeId;
+                iframeRef.current.setAttribute('loading', 'lazy');
+                iframeRef.current.setAttribute('title', title || i18n('iframe-title'));
+                iframeRef.current.frameBorder = '0';
+                iframeRef.current.scrolling = 'no';
+                iframeRef.current.width = iframeWidth;
+                iframeRef.current.style.width = iframeWidth;
+                if (iframeHeight) {
+                    iframeRef.current.style.height = iframeHeight;
+                }
+
+                container.appendChild(iframeRef.current);
+            }
+        },
+        [src, width, iframeId, name, title, height],
+    );
+
+    const handleMessage = useCallback(
+        ({data}: MessageEvent) => {
+            if (height !== 'auto' && typeof height === 'number' && iframeRef.current) {
+                iframeRef.current.height = `${height}px`;
+                return;
+            }
+
+            try {
+                const parsed = JSON.parse(data);
+                const iframeHeight = parsed['iframe-height'];
```

Comment:
Where does this message format come from? I think we are trying to keep Iframe component abstract enough

Reply:
Copy + Paste effect. Fixed

Reply:
An iframe is a kind of uniq component. You have no idea, that the content of it can be. That is the reason why we are getting data from inside of an iframe to adjust component size to its height.

Moreover, we already use this approach here https://github.com/gravity-ui/page-constructor/blob/main/src/components/YandexForm/YandexForm.tsx#L117